lib/job/arBaseJob.class.php

```diff
@@ -270,4 +270,31 @@ protected function createZipForDownload($path)
 
     return $success;
   }
+
+  /**
+   * Set job owner in user Context. ACL checks require this to be set.
+   *
+   * @param int   User ID
+   *
+   * @return null
+   */
+  protected function signIn($userId)
+  {
+    $user = QubitUser::getById($userId);
+    sfContext::getInstance()->user->signIn($user);
+  }
+
+  /**
+   * Identify this job as a background job to differentiate it from a standard
+   * CLI task. This is used to trigger ACL checks which normally short-circuit
+   * an allow full access for CLI tasks.
+   *
+   * @param array   Refence to job parameters
+   *
+   * @return null
+   */
+  protected function isBackgroundJob(&$parameters)
```

Comment:
can we change the function name here since it's a setter? The function name makes me think it's some sort of checker function that returns a bool